packages/vscode-extension/src/builders/buildIOS.ts

```diff
@@ -199,6 +203,39 @@ export async function buildIos(
   }
 }
 
+async function getOrMakeSimulator(buildConfig: IOSLocalBuildConfig) {
+  const { runtimeId } = buildConfig;
+  const devices = await listSimulators(SimulatorDeviceSet.Default); // use default location becuase xcodebuild can't use custom device sets
+  const matchingDevice = devices.find(
+    (deviceInfo) => deviceInfo.runtimeInfo?.identifier === runtimeId && deviceInfo.available
+  );
+  if (matchingDevice) {
+    Logger.info("Using existing simulator for iOS build:", matchingDevice);
+    return { simulatorUdid: matchingDevice.UDID };
```

Comment:
NIT:
```suggestion
    return { simulatorUdid: matchingDevice.UDID, cleanupSimulator: () => {} };
```
making the `cleanupSimulator` required (but noop in this case) would make calling it more readable.
You could rename it to `maybeCleanupSimulator` if you want to highlight it sometimes does nothing, but I don't think it's necessary.